### PR TITLE
[BSv5] Refactor Click-to-copy

### LINKED ISSUE #1267: Reactivate code-block copy feature
- #1245 introduced a click-to-copy button for code blocks
- Due to the issues reported in #1264, the feature had to be disabled
- Once we upgrade Bootstrap (#470), consider reactivating the click-to-copy feature.
- Other points to consider:
  - Make the feature inclusion configurable via a config param, since some projects might have and prefer to continue using their own code-copy button.
  - Add a note to the changelog when the feature is added, so that projects are more likely to find out about this nifty new feature!

Related:

- Possible feature interaction?
  - #292
  - #325
- #1266

/cc @geriom

### PR DESCRIPTION
**Pull request by geriom**
Refactor of PR #1245. Bootstrap5-compatible code.

Closes #1267

### CODE REVIEW

File: assets/scss/_code.scss [hunk 1]
```diff
@@ -1,101 +1,84 @@
 // Code formatting.
 
 .td-content {
-	// Highlighted code.
```
Comment (chalin) [inline assets/scss/_code.scss:4]: It's hard to follow what has changed in this file. Could you consider not reformatting it in this PR so that the diffs are more clearly seen?

Did you only change the lines for the c2c button?

File: CHANGELOG.md [hunk 2]
```diff
@@ -12,7 +12,11 @@ For a list of issues targeted for the next release, see the [23Q1][] milestone.
 
 **New**:
 
-**Breaking changes**:
+- **Click to copy button for Chroma-highlighted code blocks**: If you already
+  implemented this functionality on your website, you can disable it. More
+  information on the [Chroma highlighting docs][chroma-docsy].
+
+**Breaking changesdocs
```
Comment (chalin) [inline CHANGELOG.md:19]: Copy-paste error?

```suggestion
**Breaking changes**:
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -12,7 +12,11 @@ For a list of issues targeted for the next release, see the [23Q1][] milestone.
 
 **New**:
 
-**Breaking changes**:
+- **Click to copy button for Chroma-highlighted code blocks**: If you already
+  implemented this functionality on your website, you can disable it. More
+  information on the [Chroma highlighting docs][chroma-docsy].
```
Comment (chalin) [inline CHANGELOG.md:17]:
```suggestion
  implemented this functionality on your website, you can disable it. For details,
  see [Code highlighting with Chroma][chroma-docsy].
```

File: assets/scss/_code.scss [hunk 4]
```diff
@@ -1,101 +1,84 @@
...
-               color: $white;
-               padding: 2px 4px;
-               border-radius: 3px;
-               display: block;
-               visibility: hidden;
-               opacity: 0;
-               transition: visibility 0s, opacity 0.5s linear;
-            }
+      // click-to-copy button
+      button {
```
Comment (chalin) [inline assets/scss/_code.scss:23]: I'd prefer that you use a class name specific to this feature (and add the class via the js):

```suggestion
      button.td-click-to-copy {
```

File: userguide/content/en/docs/adding-content/lookandfeel.md [hunk 5]
```diff
@@ -148,6 +148,24 @@ guessSyntax: true
 {{< /tab >}}
 {{< /tabpane >}}
 
+If you are using a Docsy version later than `0.6.0`, the code blocks show a
```
Comment (chalin) [inline userguide/content/en/docs/adding-content/lookandfeel.md:151]:
```suggestion
If you are using a Docsy version `0.6.0` or later, code blocks show a
```

File: userguide/content/en/docs/adding-content/lookandfeel.md [hunk 6]
```diff
@@ -148,6 +148,24 @@ guessSyntax: true
...
+{{< tab header="hugo.toml" lang="toml" >}}
+disable_click2copy_chroma = true
+{{< /tab >}}
+{{< tab header="hugo.yaml" lang="yaml" >}}
+disable_click2copy_chroma: true
+{{< /tab >}}
+{{< tab header="hugo.json" lang="json" >}}
+"disable_click2copy_chroma": true
+{{< /tab >}}
+{{< /tabpane >}}
```
Comment (chalin) [inline userguide/content/en/docs/adding-content/lookandfeel.md:167]: I'm not a fan of the use of these manually coded TOML/YAML/JSON blocks, esp. when the configuration parameter is so simple:

```suggestion
functionality set `params.disable_click2copy_chroma` to `true` in your
configuration file.
```

File: assets/js/click-to-copy.js [hunk 7]
```diff
@@ -1,86 +1,40 @@
-let codeListings = document.querySelectorAll('.highlight > pre');
+let codeListings = document.querySelectorAll(".highlight > pre");
```
Comment (chalin) [inline assets/js/click-to-copy.js:1]: Consider running Prettier over this file, thanks: `npx prettier --single-quote --write assets/js/click-to-copy.js`

**Review by chalin — COMMENTED** [review]
Works nicely. See inline comments, thanks.

**Comment by chalin** [comment]
I've merged #1454. Apologies for the resulting merge conflicts, but as I mention in the PR, I should have done this earlier.

**Comment by chalin** [comment]
@geriom - a more general question, why not use [clipboardjs.com](https://clipboardjs.com)?

File: userguide/content/en/docs/adding-content/lookandfeel.md (hunk 5 above)
Comment (geriom) [reply userguide/content/en/docs/adding-content/lookandfeel.md:147]: This does not work for `0.6.0` and we haven't definitely decided on the next version number. I think "later than" covers that. About the article "the", although I know  I don't really need it here, I try to never omit it, based on this guideline: https://developers.google.com/style/articles

File: CHANGELOG.md (hunk 3 above)
Comment (geriom) [reply CHANGELOG.md]: Updated

File: CHANGELOG.md (hunk 2 above)
Comment (geriom) [reply CHANGELOG.md]: More like a random vim keypress :(  Fixed.

File: assets/js/click-to-copy.js (hunk 7 above)
Comment (geriom) [reply assets/js/click-to-copy.js]: Done

File: assets/scss/_code.scss (hunk 1 above)
Comment (geriom) [reply assets/scss/_code.scss]: There's nothing new in this file. I deleted all the styles for the tooltip and prettified it.

File: userguide/content/en/docs/adding-content/lookandfeel.md (hunk 6 above)
Comment (geriom) [reply userguide/content/en/docs/adding-content/lookandfeel.md]: Completely agree. I was mostly following what's already in this doc. Updated.

File: assets/scss/_code.scss (hunk 4 above)
Comment (geriom) [reply assets/scss/_code.scss]: Updated her and added the class in the js file

**Comment by geriom** [comment]
> @geriom - a more general question, why not use [clipboardjs.com](https://clipboardjs.com)?

I don't think we need an additional dependency for a straightforward feature. I can see how that library would be very helpful for single-page applications with lots of dynamic content, but here I don't see a big benefit.

**Review by chalin — APPROVED** [review]
Thanks for the updates!
